tests/DeepCopyTest/Filter/DataFilterTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace DeepCopyTest\Filter;
+
+use DeepCopy\DeepCopy;
+use DeepCopy\Filter\DataFilter;
+use DeepCopy\Matcher\PropertyMatcher;
+
+class DataFilterTest extends \PHPUnit_Framework_TestCase
+{
+    public function testApply()
+    {
+        $object = new \stdClass();
+        $object->data = array('foo' => 'bar', 'baz' => 'foo');
+
+        $cb = function($data) {
+            unset($data['baz']);
+            return $data;
+        };
+
+        $filter = new DataFilter($cb);
+
+        $this->assertEquals($cb, \PHPUnit_Framework_Assert::readAttribute($filter, 'callback'));
```

Comment:
Minor detail but I think this assertion is useless as it breaks encapsulation of the class and assume too much. If the implementation of the class changes later, the test will fail even though the class still work.

Except that the test is very good.